tf-psa-crypto/tests/suites/test_suite_psa_crypto.function

```diff
@@ -10111,28 +10114,48 @@ void generate_key(int type_arg,
         TEST_ASSUME(status != PSA_ERROR_INSUFFICIENT_MEMORY);
     }
     TEST_EQUAL(status, expected_status);
+    if (expected_status == PSA_SUCCESS) {
+        /* Test the key information */
+        PSA_ASSERT(psa_get_key_attributes(key, &got_attributes));
+        TEST_EQUAL(psa_get_key_type(&got_attributes), type);
+        TEST_EQUAL(psa_get_key_bits(&got_attributes), bits);
+
+        /* Do something with the key according to its type and permitted usage. */
+        TEST_EQUAL(mbedtls_test_psa_exercise_key(key, usage, alg, 0), 1);
+    }
+    if (!PSA_KEY_TYPE_IS_ECC(type)) {
+        expected_status = PSA_ERROR_NOT_SUPPORTED;
+    }
+
+#if !defined(MBEDTLS_ECP_RESTARTABLE)
+    expected_status = PSA_ERROR_NOT_SUPPORTED;
+#endif
+
+    status = psa_generate_key_iop_setup(&operation, &attributes);
+    TEST_EQUAL(status, expected_status);
     if (expected_status != PSA_SUCCESS) {
         goto exit;
     }
```

Comment:
This seems vaguely redundant - why not just wrap the whole iop block in #ifdef MBEDTLS_ECP_RESTARTABLE - this will also reduce the amount of tests that this will add.

Reply:
Because I want to test that calling the function when `MBEDTLS_ECP_RESTARTABLE ` is not defined returns the desired behaviour which is to return `PSA_ERROR_NOT_SUPPORTED `.

Reply:
Ok.